lib/utils/date.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:timezone/timezone.dart' as tz;
+
+DateTime? getNowInTimezone(String timezone) {
+  if (timezone.isEmpty) {
+    return null;
+  }
+  try {
+    final location = tz.getLocation(timezone);
+    return tz.TZDateTime.now(location);
+  } on tz.LocationNotFoundException {
+    return null;
+  }
+}
```

Comment:
I did try to keep an eye for this, but must have missed it (one issue with having a large commit). I'll look into making sure the IDE enforces this or warns me (from a cursory view it looks like there's a dart lint rule for this that I could take advantage of)

Reply:
Hmm neat, if there's a lint rule then we should probably just add that to `analysis_options.yaml`.

For me Android Studio also seems to avoid causing this in the first place. I actually went looking in the settings UI to try to see if there's some setting I set years ago that I can just recommend to others to set, but couldn't find one. … However, there is this line in my `options/editor.xml` (inside `~/.config/Google/AndroidStudio$version/`):
```xml
<application><!-- … -->
  <component name="EditorSettings"><!-- … -->
    <option name="IS_ENSURE_NEWLINE_AT_EOF" value="true" />
  </component>
</application>
```
That name sounds awfully like it may be the desired setting.